Resources/config/security_expressions.xml

```diff
@@ -27,7 +27,7 @@
 
     <services>
         <service id="security.extra.twig_extension" class="%security.extra.twig_extension.class%" public="false">
-            <argument type="service" id="security.context" />
+            <argument type="service" id="security.authorization_checker" />
```

Comment:
you would have to use a compiler pass to change what gets injected in the service. and your constructor should accept both the old and new classes (meaning no typehint but some instanceof check inside the constructor)